### PR TITLE
DSND-1577 Correct order of operations

### PR DESCRIPTION
**Pull request by kinpang-CH**
* Fix integration tests by calling existing steps to ensure DB changes are not done
* Behaviour tested and confirmed working on local dev environment

Resolves [DSND-1577](https://companieshouse.atlassian.net/browse/DSND-1577)

[DSND-1577]: https://companieshouse.atlassian.net/browse/DSND-1577?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java [hunk 1]
```diff
@@ -165,32 +169,28 @@ private boolean isLatestRecord(String officerId, OffsetDateTime deltaAt) {
...
+                officerId));
+
         try {
             repository.save(document);
-            savedToDb = true;
             logger.info(String.format("Disqualification is updated in MongoDb for context id: %s and officer id: %s",
-                contextId,
-                officerId));
+                    contextId,
+                    officerId));
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java:190]: Line 204 could be updated to a more functional style:
`return doc.map(DisqualificationDocument::getCreated).orElse(null);`

Line 210 can be returned inline:
`return disqualificationDocumentOptional.orElseThrow(
                () -> new IllegalArgumentException(String.format(
                "Resource not found for officer ID: %s", officerId)));`

File: src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java [hunk 2]
```diff
@@ -110,15 +110,17 @@ public void deleteDisqualification(String contextId, String officerId) {
...
         disqualifiedOfficerApiService.invokeChsKafkaApi(
-                new ResourceChangedRequest(contextId, officerId, 
+                new ResourceChangedRequest(contextId, officerId,
                         DisqualificationResourceType.CORPORATE, document.getData(), true));
         logger.info(String.format("ChsKafka api DELETED invoked updated successfully for context id: %s and officer id: %s",
+                contextId,
+                officerId));
+
+        repository.delete(document);
+        logger.info(String.format("Corporate disqualification is deleted in MongoDb officer id: %s",
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java:123]: Officer ID is being incorrectly logged as context ID here

**Review by cgregoryCH — COMMENTED** [review]
Minor comments

File: src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java (hunk 1 above)
Comment (kinpang-CH) [reply src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java:190]: Fixed in latest commit.

File: src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java (hunk 2 above)
Comment (kinpang-CH) [reply src/main/java/uk/gov/companieshouse/disqualifiedofficersdataapi/service/DisqualifiedOfficerService.java]: Probably a typo, have made it it equivalent to the Natural disqualifications delete logging message on line 146.

**Review by sthompsonCH — APPROVED** [review]

**Review by crowlandsCH — APPROVED** [review]

**Pull request merged**